documentation/AuthorizationService.md

```diff
@@ -1,123 +1,245 @@
+
+**APIs for Onboarded Solutions**
+* [addUserAuthorizedAuthorization(userId, solutionId, oauth2ClientId, selectedPreferences)](#adduserauthorizedauthorizationuserid-solutionid-oauth2clientid-selectedpreferences)
+* [getSelectedPreferences(userId, authorizationId)](#getselectedpreferencesuserid-authorizationid)
+* [getUserAuthorizedClientsForUser(userId)](#getuserauthorizedclientsforuseruserid)
+* [getUserUnauthorizedClientsForUser(userId)](#getuserunauthorizedclientsforuseruserid)
+* [revokeUserAuthorizedAuthorization(userId, authorizationId)](#revokeuserauthorizedauthorizationuserid-authorizationid)
+* [setSelectedPreferences(userId, authorizationId, selectedPreferences)](#setselectedpreferencesuserid-authorizationid-selectedpreferences)
+
+**APIs for Privileged Preferences Creators**
+* [getAuthorizationByAccessToken(accessToken)](#getauthorizationbyaccesstokenaccesstoken)
+* [grantPrivilegedPrefsCreatorAuthorization(clientId, scope)](#grantprivilegedprefscreatorauthorizationclientid-scope)
+* [revokePrivilegedPrefsCreatorAuthorization(authorizationId)](#revokeprivilegedprefscreatorauthorizationauthorizationid)
+
+**Note**: Privileged Preferences Creator Clients are authorized using [OAuth2 Client Credentials Grant](https://wiki.gpii.net/w/GPII_OAuth_2_Guide#Client_Credentials_Grant).
+
+**APIs for Web Preferences Consumers**
+* [addUserAuthorizedAuthorization(userId, solutionId, oauth2ClientId, selectedPreferences)](#adduserauthorizedauthorizationuserid-solutionid-oauth2clientid-selectedpreferences)
+* [exchangeCodeForAccessToken(code, clientId, redirectUri)](#exchangecodeforaccesstokencode-clientid-redirecturi)
+* [getAuthorizationByAccessToken(accessToken)](#getauthorizationbyaccesstokenaccesstoken)
+* [getSelectedPreferences(userId, authorizationId)](#getselectedpreferencesuserid-authorizationid)
+* [getUserAuthorizedClientsForUser(userId)](#getuserauthorizedclientsforuseruserid)
+* [getUserUnauthorizedClientsForUser(userId)](#getuserunauthorizedclientsforuseruserid)
+* [grantWebPrefsConsumerAuthCode(userId, clientId, redirectUri, selectedPreferences)](#grantwebprefsconsumerauthcodeuserid-clientid-redirecturi-selectedpreferences)
+* [revokeUserAuthorizedAuthorization(userId, authorizationId)](#revokeuserauthorizedauthorizationuserid-authorizationid)
+* [setSelectedPreferences(userId, authorizationId, selectedPreferences)](#setselectedpreferencesuserid-authorizationid-selectedpreferences)
+* [userHasAuthorizedWebPrefsConsumer(userId, clientId, redirectUri)](#userhasauthorizedwebprefsconsumeruserid-clientid-redirecturi)
+
+**Note**: Web Preferences Consumer Clients are authorized using [OAuth2 Authorization Code Grant](https://wiki.gpii.net/w/GPII_OAuth_2_Guide#Authorization_Code_Grant).
+
+#### addUserAuthorizedAuthorization(userId, solutionId, oauth2ClientId, selectedPreferences)
```

Comment:
It seems that throughout the implementation, the parameter "oauth2ClientId" has been renamed to "clientId" - should we update these docs to reflect this?

Reply:
`oauth2ClientId` hasn't been renamed to `clientId`. Right now, 4 client types handled by GPII can be categorized into 2 groups:

1. OAuth2 clients including GPII app installations, privileged prefs creators and web prefs consumers. These clients are authorized via OAuth2;
2. Non-OAuth2 client that is onboarded solutions. Users are able to define privacy profiles for this type of client but this client type doesn't go thru OAuth2 authorization at all.

Both of these 2 groups use `clientId` in generally. But when it drills down to specific groups, `oauth2ClientId` and `solutionId` are used.